wandb/sdk/artifacts/artifact_saver.py

```diff
@@ -137,7 +141,7 @@ def _save_internal(
         if self._server_artifact["state"] == "COMMITTED":
             if use_after_commit:
                 self._api.use_artifact(artifact_id)
-            return self._server_artifact
+            return self._server_artifact, None
```

Comment:
Nit: returning a fulfilled `Future` here would make the interface easier:
```suggestion
            result = concurrent.futures.Future()
            result.set_result(None)
            return self._server_artifact, result
```

Reply:
I spent some time looking into this change, I'm leaning towards not doing it. The main reason is I couldn't find an easy way to create a complete future. One potential is I can do:

```
future.set_result(None)
return future
```
but the doc string says: 

```
    def set_result(self, result):
        """Sets the return value of work associated with the future.

        Should only be used by Executor implementations and unit tests.
        """
```

and that makes me a bit nervous.